src/transactionUtils.ts

```diff
@@ -17,3 +18,165 @@ export function getMaxTsWithBuffer(transactions: SlotTransactions) {
   );
   return (max(txnTs) ?? 0) + chartBufferMs;
 }
+
+export function getTxnStateDurations(
+  transactions: SlotTransactions,
+  txnIdx: number,
+  bundleTxnIdx: number[] | undefined,
+  client: "Frankendancer" | "Firedancer",
```

Comment:
```suggestion
  client: Client,
```

we got a zod inferred type